runtime/experimental/makefiles/curl.mk

```diff
@@ -0,0 +1,45 @@
+SHELL := /bin/bash
+.DEFAULT_GOAL := make_curl
+build_dir_curl = ${DEPS}/curl
+
+fetch_curl:
+	mkdir -p ${build_dir_curl}
+	/usr/bin/git clone https://github.com/curl/curl.git ${build_dir_curl}
+	cd ${build_dir_curl} && \
+	/usr/bin/git checkout curl-$(subst .,_,${VERSION_CURL})
+
+configure_curl:
+	cd ${build_dir_curl} && \
+	${build_dir_curl}/buildconf && \
+	${build_dir_curl}/configure \
+        --prefix=${TARGET} \
+		--with-ca-bundle=${CA_BUNDLE} \
+        --enable-shared \
+        --disable-static \
+		--enable-optimize \
+		--disable-warnings \
+        --disable-dependency-tracking \
+		--with-zlib \
+		--enable-http \
+		--enable-ftp  \
+		--enable-file \
+		--enable-ldap \
+		--enable-ldaps  \
+		--enable-proxy  \
+		--enable-tftp \
+		--enable-ipv6 \
+		--enable-openssl-auto-load-config \
+		--enable-cookies \
+		--with-gnu-ld \
+		--with-ssl \
+		--with-libssh2
+
+build_curl:
+	cd ${build_dir_curl} && \
+	$(MAKE) install
```

Comment:
Why is the `curl` extension (and other similar extensions) compiled from source and not installed via `--with-curl` like [here](https://github.com/mnapoli/bref/blob/ebdf5331deadb2d7a2a794b9175430026dc7d779/runtime/default/build.sh#L56-L69)?

Reply:
When php-curl is built against a different openssl library than PHP itself is built against, that complicates things.

In order to have PHP built against a curl library that is built against a sane Openssl version, we have to build it ourselves. For developers who have legal compliance requirements, this is important.

Building against the php-curl extension against the latest curl ensures that our developers have access to the latest curl feature-set, as well as the collective bugfixes.

**Technical Details**
That `--with-curl` is actually building against the old system curl ([version 7.51.0, released Nov 3, 2016](https://github.com/curl/curl/releases?after=curl-7_52_0)), those shared libraries/executables are already installed on the system by default, and the development files are being add with 
 https://github.com/mnapoli/bref/blob/ebdf5331deadb2d7a2a794b9175430026dc7d779/runtime/default/build.sh#L29

A cursory glance at https://curl.haxx.se/changes.html#7_52_0 through https://curl.haxx.se/changes.html#7_63_0 looks like over 1,000 bugfixes and perhaps a hundred or so other changes.

Additionally, the system curl is compiled against [OpenSSL 1.0.1k-fips 8 Jan 2015](https://github.com/openssl/openssl/releases?after=OpenSSL_0_9_8ze) as well as a number of other old libs (zlib, libssh2) that differ from the version we will compile PHP against.

Reply:
Awesome thank you